### DISCUSSION TITLE
import Raystation SMLC plan and recalculate and compare matrad vs Raystation doses

### DISCUSSION THREAD

**Discussion by gleesoi** [e0404/matRad#705] [General]
i want to recalc my SMLC IMRT Raystation plan in matrad and compare doses if possible? i tried the below using my imported plan but it obviously didnt find my agility machine file so i changed to Generic machine and tried to compute. 2 beams appeared to compute ok and then failed at beam3 with below error. Any help much appreciated to achieve my aims, thanks!

Also wondering if possible a centres specific beam model can be used to more closely match matrad and what parameters are needed& could be altered?

![image](https://github.com/e0404/matRad/assets/75881142/41fe4046-b186-4b76-bea8-73ef2bcab93f)

**Comment by gleesoi** [comment]
as a follow up i imported instead a 2 field conformal plan using below code/

%setup the path & configuration and tell you the current version
matRad_rc %Execute in base folder of matRad

matRadGUI

[allfiles,patients] = matRad_scanDicomImportFolder("U:\Ian Gleeson\MATRAD\all data test"); %MatRad will also be able to separate multiple patients, but this example will only work if there's only a single patient in the folder.
ctFiles = strcmp(allfiles(:,2),'CT');
rtssFiles = strcmpi(allfiles(:,2),'rtstruct'); %note we can have multiple RT structure sets, matRad will always import the first it finds
rtplanFiles = strcmpi(allfiles(:,2),'rtplan');

importFiles.ct = allfiles(ctFiles,1);%All CT slice filepaths stored in a cell array like {'CTSlice1.dcm','CTSlice2.dcm'}; 
importFiles.rtss = allfiles(rtssFiles,1); %will also be a cell array like {'RTStruct.dcm'}; 
importFiles.rtplan = allfiles(rtplanFiles,1);

%Use the first ct file to get resolution
dcmInfoCt = dicominfo(importFiles.ct{1}); 
importFiles.resx = dcmInfoCt.PixelSpacing(1);
importFiles.resy = dcmInfoCt.PixelSpacing(2);
importFiles.resz = dcmInfoCt.SliceThickness; %some CT dicoms do not follow the standard and use SpacingBetweenSlices

%We need to set one more variable I forgot to mention above
importFiles.useDoseGrid = false;

[ct,cst,pln] = matRad_importDicom(importFiles);

%needs bixels first as doesnt recognise 'field' as input
pln.propStf.bixelWidth      = 10;

%change machine to genric as cant find agility base data / machine file from raystattion imported plan
pln.machine = 'Generic';

%Generate Beam Geometry STF
stf = matRad_generateStf(ct,cst,pln);   %fails as cant find machine file agility so thats why we change to generic above


now when i want to calc the beams and see dose i tried below but error as no weight found?

% calculate direct??
resultGUI = matRad_calcDoseDirect(ct,stf,pln,cst); 

% need shape info in stf?  Error using matRad_calcDoseDirect (line 39) No weight vector available. Please provide w or add info to stf
 
 
![image](https://github.com/e0404/matRad/assets/75881142/33c3f75f-33b3-4aa6-9b65-743a5e070427)

![image](https://github.com/e0404/matRad/assets/75881142/7fae2090-8e54-4519-aa6b-37faa53c372b)

**Comment by edwardwang1** [comment]
Hi @gleesoi,

Not sure if you're still working on this, but I have imported a multi-beam IMRT plan from Eclipse into matRad, and recalculated the dose. I actually did everything from the GUI (first importing the DICOM, then loading the mat file). Afterwards, I simply clicked the Recalc button.
![image](https://github.com/user-attachments/assets/a18a7771-623b-425d-b0d8-b891b19c30b3)

**Reply by Little-Stars520** [reply]
Hi,@edwardwang1,
In order to verify the matRad dose calculation of our center's machine (we use varian EDGE, equipped with HD120), do we need to generate the pencil beam kernel using the commission data of our center's machine? If I want to change the properties of the MLC, such as tongue-and-groove width, DLG, radius, etc., and then generate the dose of our center's machine under the new MLC, how should we make the changes?

**Reply by wahln** [reply]
Yes, you need to use the commissioning data. 
As said in #263, changes in the source / MLC need a new kernel dataset from new measurements (or MC simulations). Pencil-beam algorithms are not really flexible in that regard, new beamline components need new kernels.
I think this would be best possible with a Monte Carlo tool - what this would need is:
- a (virtual) source model for the machine (open fields)
- a geometric model of the MLC
There's some possibilities to include MLCs in our TOPAS interface, but the virtual source model you would need to derive from your commissioning data.
